manager/orchestrator/constraintenforcer/constraint_enforcer.go

```diff
@@ -83,10 +84,11 @@ func (ce *ConstraintEnforcer) rejectNoncompliantTasks(node *api.Node) {
 		log.L.WithError(err).Errorf("failed to list tasks for node ID %s", node.ID)
 	}
 
-	var availableMemoryBytes, availableNanoCPUs int64
+	available := &api.Resources{}
+	available.ThirdParty = make(map[string]*api.ThirdPartyResource)
```

Comment:
Moving this to an `else` clause would avoid the map allocation if we are making a copy of the node's resources anyway.